### PR TITLE
Tests. Organize and modernize helper tests

### PR DESCRIPTION
**Pull request by grummbeer**
Reorganize the tests for helper

Issue [#34379](https://mantisbt.org/bugs/view.php?id=34379)

- Move the file to /Helper.
- Rename the file to reflect what it tests.
- Fix doc block annoation (but removed with the last commit)
- PHP 7.4 Update array syntax from `array()` to `[]`
- PHP 7.4 Add type hints and add `declare(strict_types=1);`
- PHP 7.4 Use the Generator syntax to yield the samples instead of creating an array first. 
- Change `assertEquals` to `assertSame` to be more strict.

```
vendor/bin/phpunit tests/Mantis/Helper/ArrayTransposeTest.php
PHPUnit 9.6.18 by Sebastian Bergmann and contributors.

.......                                                             7 / 7 (100%)
```

The last commit is "Remove unnecessary doc block annotations". Can be dropped if not wanted.

```php
/**
* Tests helper_array_transpose() with invalid values.
*
* @param array $p_in  Input value.
* ^^^^^^ Removed, it is self descriptive
* @return void
* ^^^^^^^ Removed, it is self descriptive
*
* @dataProvider providerArrayTransposeInvalid
*/
public function testArrayTransposeInvalid( array $p_in ): void {}
                                           ^^^^^^^^^^^    ^^^^^
                                                |           | Return type clear
                                                ^
                    Type of param is clear, it’s value is not explicitly stated, 
                             but there is no doubt what it is. 
```

```
/**
 * @return Generator List of test cases
 * ^^^^^^^ Removed, it is self descriptive
 */
public function providerArrayTransposeValid(): Generator{
```

If these changes are welcome, I will do it also for the remaining tests in test/Mantis.

### CODE REVIEW

File: tests/Mantis/Helper/ArrayTransposeTest.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
```
Comment (dregad) [inline tests/Mantis/Helper/ArrayTransposeTest.php:3]: Move below header comments + coding guidelines

**Review by dregad — CHANGES_REQUESTED** [review]
### Remove unnecessary doc block annotations

>\* @param array $p_in  Input value.
>\* ^^^^^^ Removed, it is self descriptive

While I agree that this specific example is somewhat contrived and does not really add any value, I prefer to keep those @param directives as they allow documenting what the test case expects, which is useful when using a @dataProvider

>\* @return void
>\* ^^^^^^^ Removed, it is self descriptive
>
> \* @return Generator List of test cases
> \* ^^^^^^^ Removed, it is self descriptive

Agreed, useless and redundant with newly added return type on function declaration.

### Commit messages

A general comment regarding commit messages, please read https://mantisbt.org/wiki/doku.php/mantisbt:git_commit_messages for guidelines. In particular, 

- try to avoid commit 1st line longer than 50(ish) chars); systematic `Tests.` prefix is not really necessary
- reference bugtracker issue for auto-linking (although when submitting a series of commits like this, it is not strictly necessary as the reference can be added in the merge commit when pulling the branch in)

**Comment by dregad** [comment]
Note: I reviewed GetLinkAttributesTest.php in #1983. Not sure why you submitted it as a separate PR.

**Comment by grummbeer** [comment]
> Note: I reviewed GetLinkAttributesTest.php in https://github.com/mantisbt/mantisbt/pull/1983. Not sure why you submitted it as a separate PR.

That was unintentional … sorry, should I reopen another?

I left it open because I thought if the other one will accepted, this one can merge on top.

File: tests/Mantis/Helper/ArrayTransposeTest.php (hunk 1 above)
Comment (grummbeer) [reply tests/Mantis/Helper/ArrayTransposeTest.php:3]: you mean the array style 

`[ [ 'a', 'b', ] ]` to `[['a', 'b']]`?

**Comment by grummbeer** [comment]
> Commit messages
> A general comment regarding commit messages, please read https://mantisbt.org/wiki/doku.php/mantisbt:git_commit_messages for guidelines. In particular,

> try to avoid commit 1st line longer than 50(ish) chars); systematic Tests. prefix is not really necessary
> reference bugtracker issue for auto-linking (although when submitting a series of commits like this, it is not strictly necessary as the reference can be added in the merge commit when pulling the branch in)

Thank you, will follow.

**Comment by dregad** [comment]
> should I reopen another?

No, don't worry about it. Just commit further fixes to the *test-get-link-attributes* branch, this way I can finish review and merge #1983, and as you said *tests-organize-helper-tests* branch can be rebased on top afterwards.

File: tests/Mantis/Helper/ArrayTransposeTest.php (hunk 1 above)
Comment (grummbeer) [reply tests/Mantis/Helper/ArrayTransposeTest.php:3]: I am fiddeling with the ecs and your sniffs, 

> Arrays elements should be referenced without spaces, e.g. $arr['index'] not $arr[ 'index' ]

Hopefully such will fixed with the "ArraySyntax" fixer in the future.

Comment (grummbeer) [reply tests/Mantis/Helper/ArrayTransposeTest.php:3]: array style fixed.

**Comment by grummbeer** [comment]
damn … messed rebase.please wait for "ready".

> While I agree that this specific example is somewhat contrived and does not really add any value, I prefer to keep those @param directives as they allow documenting what the test case expects, which is useful when using a @dataProvider

Agree in general, but a description like "input array" is not very helpful either. Especially in Tests, the values are not hard to recognize.

File: tests/Mantis/Helper/ArrayTransposeTest.php (hunk 1 above)
Comment (dregad) [reply tests/Mantis/Helper/ArrayTransposeTest.php]: > you mean the array style

Actually I meant the spacing around parentheses and equal sign `declare( strict_types = 1 );`

But you did right with the arrays too

**Comment by grummbeer** [comment]
ready for review.

**Review by dregad — APPROVED** [review]
Looks good. 

I just don't like squashing move + rename + code changes in a single commit as it breaks continuity in Git history.

I'll split the move and rename operations into separate commits to avoid that.

**Comment by grummbeer** [comment]
> I just don't like squashing move + rename + code changes in a single commit

OK. Still learning.

**Pull request merged**